src/components/select/select.js

```diff
@@ -0,0 +1,216 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Select as PfSelect, SelectOption as PfSelectOption, SelectVariant } from '@patternfly/react-core';
+import _cloneDeep from 'lodash/cloneDeep';
+import _isEqual from 'lodash/isEqual';
+import _isPlainObject from 'lodash/isPlainObject';
+import { helpers } from '../../common/helpers';
+
+class Select extends React.Component {
+  state = { isExpanded: false, options: null, selected: null };
+
+  componentDidMount() {
+    const { options } = this.state;
+
+    if (options === null) {
+      this.formatOptions();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { options } = this.props;
+
+    if (!_isEqual(prevProps.options, options)) {
+      this.formatOptions();
+    }
+  }
+
+  onSelect = (event, titleSelection) => {
+    const { options } = this.state;
+    const { id, name, onSelect, variant } = this.props;
+
+    const updatedOptions = options;
+    const optionsIndex = updatedOptions.findIndex(option => option.title === titleSelection);
+    updatedOptions[optionsIndex].selected =
+      variant === SelectVariant.single ? true : !updatedOptions[optionsIndex].selected;
+
+    if (SelectVariant.single) {
+      updatedOptions.forEach((option, index) => {
+        if (optionsIndex !== index) {
+          updatedOptions[index].selected = false;
+        }
+      });
+    }
+
+    const updateSelected =
+      variant === SelectVariant.single
+        ? titleSelection
+        : updatedOptions.filter(opt => opt.selected === true).map(opt => opt.title);
+
+    this.setState(
+      {
+        options: updatedOptions,
+        selected: updateSelected
+      },
+      () => {
+        const mockUpdatedOptions = _cloneDeep(updatedOptions);
+
+        const mockTarget = {
+          id,
+          name: name || id,
+          value: mockUpdatedOptions[optionsIndex].value,
+          selectedIndex: optionsIndex,
+          type: `select-${(variant === SelectVariant.single && 'one') || 'multiple'}`,
+          options: mockUpdatedOptions
+        };
+
+        const mockEvent = {
+          ...mockTarget,
+          target: { ...mockTarget },
+          currentTarget: { ...mockTarget },
+          persist: helpers.noop
+        };
+
+        onSelect({ ...mockEvent }, optionsIndex, mockUpdatedOptions);
```

Comment:
this is the kind of API response I would expect from a generic Dropdown/Select/OptionsMenu - if an object of options passed, it should be returned to the consumer as an object (that can they can access it again if needed).

Major kudos on this! Would like to see the same sort of behavior in PF some day! 👍